File: cdk/python/sqs-queue/sqs_queue/sqs_queue_stack.py
from aws_cdk import (
    # Duration,
    Stack,
    aws_sqs as sqs,
)
from constructs import Construct

class SqsQueueStack(Stack):

    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        # SQS Queue that passes linting and security checks
        queue = sqs.Queue(
            self,
            "test-queue-cw-iac-examples",
            encryption=sqs.QueueEncryption.KMS_MANAGED,
            enforce_ssl=True,
        )

File: cdk/typescript/sqs-queue/lib/sqs-queue-stack.ts
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as sqs from 'aws-cdk-lib/aws-sqs';

export class SqsQueueStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // SQS queue that passes linting and security checks
    new sqs.Queue(this, 'ExampleSqsQueue', {
        encryption: sqs.QueueEncryption.KMS_MANAGED,
        enforceSSL: true,
    });
  }
}
